spi/src/main/java/com/flipkart/varadhi/spi/authn/AuthenticationProvider.java

```diff
@@ -0,0 +1,13 @@
+package com.flipkart.varadhi.spi.authn;
+
+import com.flipkart.varadhi.entities.auth.UserContext;
+import com.flipkart.varadhi.spi.ConfigFileResolver;
+import io.vertx.core.Future;
+
+import java.net.URI;
+import java.util.Map;
+
+public interface AuthenticationProvider {
+    Future<Boolean> init(ConfigFileResolver resolver, AuthenticationOptions authenticationOptions);
```

Comment:
for now I think we can omit the resolver parameter. Why is it needed?

Reply:
yes. this can be removed.